collated/main/khooroko.md

```diff
@@ -0,0 +1,1424 @@
+``` java
+    /**
+     * Binds the individual UI elements to observe their respective {@code Person} properties
+     * so that they will be notified of any changes.
+     */
+    private void bindListeners(ReadOnlyPerson person) {
+        name.textProperty().bind(Bindings.convert(person.nameProperty()));
+        handphone.textProperty().bind(Bindings.convert(person.handphoneProperty()));
+        homePhone.textProperty().bind(Bindings.convert(person.homePhoneProperty()));
+        officePhone.textProperty().bind(Bindings.convert(person.officePhoneProperty()));
+        address.textProperty().bind(Bindings.convert(person.addressProperty()));
+        postalCode.textProperty().bind(Bindings.convert(person.postalCodeProperty()));
+        cluster.textProperty().bind(Bindings.convert(person.clusterProperty()));
+        email.textProperty().bind(Bindings.convert(person.emailProperty()));
+        debt.textProperty().bind(Bindings.convert(person.debtProperty()));
+        totalDebt.textProperty().bind(Bindings.convert(person.totalDebtProperty()));
+        interest.textProperty().bind(Bindings.convert(person.interestProperty()));
+        dateBorrow.textProperty().bind(Bindings.convert(person.dateBorrowProperty()));
+        deadline.textProperty().bind(Bindings.convert(person.deadlineProperty()));
+        dateRepaid.textProperty().bind(Bindings.convert(person.dateRepaidProperty()));
+        //TODO: fix tag colours. person.tagProperty().addListener((observable, oldValue, newValue) -> {
+        tags.getChildren().clear();
+        initTags(person);
+    }
+
+    /**
+     * Initializes and styles tags belonging to each person.
+     * @param person must be a valid person.
+     */
+    private void initTags(ReadOnlyPerson person) {
+        person.getTags().forEach(tag -> {
+            Label tagLabel = new Label(tag.tagName);
+            tagLabel.setStyle("-fx-font-size:" + "15px");
+            tags.getChildren().add(tagLabel);
+        });
+        logger.finest("All tags for " + person.getName().toString() + " initialized in info");
+    }
+
+    /**
+     * Sets all info fields to not display anything.
+     */
+    private void loadDefaultPage() {
+        Label label;
+        Text text;
+        for (Node node: pane.getChildren()) {
+            if (node instanceof Label) {
+                label = (Label) node;
+                label.setText("");
+            } else if (node instanceof Text) {
+                text = (Text) node;
+                text.setText("");
+            } else if (node instanceof TextFlow) {
+                for (Node subNode: ((TextFlow) node).getChildren()) {
+                    if (subNode instanceof Text) {
+                        text = (Text) subNode;
+                        text.setText("");
+                    }
+                    if (subNode instanceof Label) {
+                        label = (Label) subNode;
+                        label.setText("");
+                    }
+                }
```

Comment:
Avoid arrow-head code, perhaps abstracting the for loop into a method will imrpvoe this.